readline/buffer.go

```diff
@@ -17,12 +18,28 @@ type Buffer struct {
 	Height    int
 }
 
-func NewBuffer(prompt *Prompt) (*Buffer, error) {
-	fd := int(os.Stdout.Fd())
-	width, height, err := term.GetSize(fd)
+func getTermSize() (width, height int, err error) {
+	// if we can't get the size from stderr, try /dev/tty
+	fd := int(os.Stderr.Fd())
+	width, height, err = term.GetSize(fd)
 	if err != nil {
-		fmt.Println("Error getting size:", err)
-		return nil, err
+		tty, ttyErr := os.OpenFile("/dev/tty", os.O_RDWR, 0)
+		if ttyErr != nil {
+			return 0, 0, errors.New("Error getting term size: " + ttyErr.Error())
+		}
+		defer tty.Close()
+		width, height, err = term.GetSize(int(tty.Fd()))
+		if err != nil {
+			return 0, 0, errors.New("Error getting term size: " + err.Error())
+		}
+	}
+	return width, height, nil
+}
```

Comment:
A small recommendation
```go
func getTermSize() (width, height int, err error) {
        // try getting size from stdout
        fd := int(os.Stdout.Fd())
        width, height, err := term.GetSize(fd)
        if err == nil {
                return width, height, nil
        }

        // try stderr
        fd := int(os.Stderr.Fd())
        width, height, err = term.GetSize(fd)
        if err == nil {
                return width, height, nil
        }

        // try /dev/tty
        tty, ttyErr := os.OpenFile("/dev/tty", os.O_RDWR, 0)
        defer tty.Close()
        if ttyErr == nil {
                width, height, err = term.GetSize(int(tty.Fd()))
                if err == nil {
                        return width, height, nil
                }
        }

        // return a default width and height
        return 80, 24, errors.New("Using default width: 80 and height: 24 Error getting term size: " + ttyErr.Error())
}
```
This way we lower the indentation amount and return a default. i am a beginner go programmer so i apologize if this is not idiomatic go code.

Reply:
Thank you for the suggestion. In the latest commit I use `errors.Join` to combine all possible errors and the method will try stdout, stderr and /dev/tty in order.